homeassistant/components/cloud/helpers.py

```diff
@@ -29,3 +43,237 @@ def _get_logs() -> list[str]:
+        parameters = convert(tool.parameters, custom_serializer=custom_serializer)
+
+        spec: FunctionToolParam = {
+            "type": "function",
+            "name": tool.name,
+            "strict": False,
+            "description": tool.description,
+            "parameters": parameters,
+        }
+
+        return spec
+
+
+class LLMFileHelper:
+    """Helper methods for LLM file handling."""
+
+    @staticmethod
+    def _convert_image_for_editing(data: bytes) -> tuple[bytes, str]:
+        """Ensure the image data is in a format accepted by OpenAI image edits."""
+        stream = io.BytesIO(data)
+        with Image.open(stream) as img:
+            mode = img.mode
+            if mode not in ("RGBA", "LA", "L"):
+                if mode == "RGB":
+                    img = img.convert("RGBA")
+                else:
+                    img = img.convert("RGBA")
+
+            output = io.BytesIO()
+            if img.mode in ("RGBA", "LA"):
+                img.save(output, format="PNG")
+                return output.getvalue(), "image/png"
+            if img.mode == "L":
+                img.save(output, format="PNG")
+                return output.getvalue(), "image/png"
+
+            img.save(output, format=img.format or "PNG")
+            return output.getvalue(), f"image/{(img.format or 'png').lower()}"
+
+    @staticmethod
+    async def async_prepare_image_generation_attachments(
+        hass: HomeAssistant, attachments: list[conversation.Attachment]
+    ) -> list[LLMImageAttachment]:
+        """Load attachment data for image generation."""
+
+        def prepare() -> list[LLMImageAttachment]:
+            items: list[LLMImageAttachment] = []
+            for attachment in attachments:
+                if not attachment.mime_type or not attachment.mime_type.startswith(
+                    "image/"
+                ):
+                    raise HomeAssistantError(
+                        "Only image attachments are supported for image generation"
+                    )
+                path = attachment.path
+                if not path.exists():
+                    raise HomeAssistantError(f"`{path}` does not exist")
+
+                data = path.read_bytes()
+                mime_type = attachment.mime_type
+
+                try:
+                    data, mime_type = LLMFileHelper._convert_image_for_editing(data)
+                except HomeAssistantError:
+                    raise
+                except Exception as err:
+                    raise HomeAssistantError(
+                        "Failed to process image attachment"
+                    ) from err
+
+                items.append(
+                    LLMImageAttachment(
+                        filename=path.name,
+                        mime_type=mime_type,
+                        data=data,
+                    )
+                )
+
+            if attachments and len(items) == 1:
```

Comment:
What is this if statement doing, can you add a comment why duplicate an attachment?